### DISCUSSION TITLE
Maven Polyglot Scala

### DISCUSSION THREAD

**Discussion by lefou** [inkytonik/kiama#36] [Show and tell]
Hello 👏‍,

I just wanted to say "hi" and make you aware that `polyglot-scala`, a Maven extension which allows you to write Maven project in Scala instead of XML, is using kiama under the hood.[^1]

https://github.com/takari/polyglot-maven/tree/master/polyglot-scala

The project exists for many years, and I started to maintain the Scala dialect in 2015. Today, I migrated it from Scala 2.11 to [2.12](https://github.com/takari/polyglot-maven/pull/277) and then to [2.13](https://github.com/takari/polyglot-maven/pull/278). As part of it, I needed to bump the kiama dependency and really appreciated the nicely written changelog. Kudos to whoever felt the need to properly list all changing details in this document. This is rarely seen in other Scala libraries.

[^1]: Don't read this as advertisement for Maven, although I can really recommend it when you are stuck with Maven. If you are in need for a nice Java / Scala build tool, I can really recommend [Mill](https://github.com/com-lihaoyi/mill/), which I also maintain.

**Comment by inkytonik** [comment]
Hi Tobias,

Thanks very much for your message. I've always wondered if anyone read those release notes, so it's great to find out that at least one person does! And it's also fun to hear of places where Kiama is used. The nature of open source is that we often don't know much about who is using what we put out there. 

I haven't had much chance to work in Scala for a while, since I changed jobs and was working Haskell for a few years and mainly C++ now. So I've lost close contact with the Scala world. I wonder how the move (or otherwise) to Scala 3 is going. Have you used it much?

regards,
Tony Sloane
